test/test_profile.py

```diff
@@ -159,3 +162,75 @@ def execute(
             """
         )
         assert result == self.expected_value
+
+
+@pytest.mark.parametrize(
+    "seed_tar",
+    [None, Path("profile.tar.gz")],
+    ids=["without_seed_tar", "with_seed_tar"],
+)
+@pytest.mark.parametrize("stateful", [True, False], ids=["stateful", "stateless"])
+@pytest.mark.parametrize(
+    "testcase",
+    ["on_normal_operation", "on_crash", "on_crash_during_launch", "on_timeout"],
+)
```

Comment:
I wonder if the `stateless-without_seed_tar` test cases are adding any value. @vringar, what do you think? If we decide that they are redundant, I can tweak the parametrization and we'll have 4 tests less.

Reply:
I'd say they are redundant, so I'm going to adjust the parametrization and create a PR against your branch tomorrow.

Reply:
Thanks for confirming! I could push this now since it's a pretty trivial change. Shall I?

Reply:
Done!

Reply:
Thanks so much!